### PR TITLE
Product filters - Step 1: UI for filters

### PR DESCRIPTION
**Pull request by anitaa1990**
Addresses step 1 of #2117 by adding the UI for filter list and filter sub list. This PR does not update the product list once filters are selected, it only addresses the UI for filters screen. This will take place in a subsequent PR.

**Note that this feature is only enabled for debug users and this PR is to be merged to a feature branch.** 

#### Screenshots
. . . . . . . . . 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/layout/fragment_product_filter_child_list.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```
[the remark this answers is not in the document]
Comment (anitaa1990) [reply WooCommerce/src/main/res/layout/fragment_product_filter_child_list.xml]: 👍 Fixed in c7a61f5

File: WooCommerce/src/main/res/layout/product_filter_list_item.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    style="@style/Woo.Card.WithoutPadding"
```
[the remark this answers is not in the document]
Comment (anitaa1990) [reply WooCommerce/src/main/res/layout/product_filter_list_item.xml]: 👍 Fixed in c7a61f5

File: WooCommerce/src/main/res/layout/product_filter_child_list_item.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    style="@style/Woo.Card.WithoutPadding"
```
[the remark this answers is not in the document]
Comment (anitaa1990) [reply WooCommerce/src/main/res/layout/product_filter_child_list_item.xml]: 👍 Fixed in c7a61f5

File: WooCommerce/src/main/res/layout/fragment_product_filter_list.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    <View
+        style="@style/Woo.Divider"
+        android:layout_marginStart="@dimen/major_100"
+        app:layout_constraintBottom_toTopOf="@+id/filterList_btnFrame"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/filterList"
+        app:layout_constraintVertical_bias="0.0" />
+
+    <com.google.android.material.card.MaterialCardView
```
[the remark this answers is not in the document]
Comment (anitaa1990) [reply WooCommerce/src/main/res/layout/fragment_product_filter_list.xml]: 👍 Fixed in c7a61f5

**Comment by anitaa1990** [comment]
> it took me a minute to figure out the relationship between the ProductFilterList and ProductFilterChildList. I'm wondering if it would make more sense to use something like ProductFilterList and ProductFilterOptionList?

Fixed in 6b537dd

Ready for another round 👍

**Comment by AmandaRiu** [comment]
Also there are merge conflicts.

**Comment by anitaa1990** [comment]
Thanks for the review @AmandaRiu! I fixed the merge conflicts in 6fd9507 and modified some child values to option in bb8f347. There are still some pending from methods that are no longer used in the next PR so I've left them alone.

Ready for another round 👍

File: WooCommerce/src/main/res/layout/fragment_product_filter_list.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (AmandaRiu) [reply WooCommerce/src/main/res/layout/fragment_product_filter_list.xml]: Looks like you missed this one...

File: WooCommerce/src/main/res/layout/product_filter_list_item.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:minHeight="@dimen/min_tap_target"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_filter_list_item.xml:8]: My apologies, I didn't notice this until I was retesting, but this list isn't supporting the click "ripple" effect. I suggest adding `android:background="?attr/selectableItemBackground"` here.

File: WooCommerce/src/main/res/layout/product_filter_option_list_item.xml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:minHeight="@dimen/min_tap_target"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_filter_option_list_item.xml:8]: My apologies, I didn't notice this until I was retesting, but this list isn't supporting the click "ripple" effect. I suggest adding `android:background="?attr/selectableItemBackground"` here.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductType.kt [hunk 7]
```diff
@@ -1,11 +1,15 @@
...
+import androidx.annotation.StringRes
+import com.woocommerce.android.R
+import org.wordpress.android.fluxc.network.rest.wpcom.wc.product.CoreProductType
+
+enum class ProductType(@StringRes val stringResource: Int = 0, val value: String = "") {
+    SIMPLE(R.string.product_type_simple, CoreProductType.SIMPLE.value),
+    GROUPED(R.string.product_type_grouped, CoreProductType.GROUPED.value),
+    EXTERNAL(R.string.product_type_external, CoreProductType.EXTERNAL.value),
+    VARIABLE(R.string.product_type_variable, CoreProductType.VARIABLE.value),
+    VARIATION(R.string.product_type_variation, CoreProductType.VARIABLE.value);
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductType.kt:12]: VARIATION is set to "variable" -- should be set to `CoreProductType.VARIATION.value`

File: WooCommerce/src/main/res/layout/product_filter_option_list_item.xml (hunk 6 above)
Comment (anitaa1990) [reply WooCommerce/src/main/res/layout/product_filter_option_list_item.xml:8]: Thanks Amanda! Addressed in the next PR (`issue/2117-step-2-filter-products`)

File: WooCommerce/src/main/res/layout/product_filter_list_item.xml (hunk 5 above)
Comment (anitaa1990) [reply WooCommerce/src/main/res/layout/product_filter_list_item.xml:8]: Thanks Amanda! Addressed in the next PR (`issue/2117-step-2-filter-products`)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductType.kt (hunk 7 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductType.kt:12]: This is addressed in the next PR and requires a change in FluxC first

**Pull request merged**